ephios/core/services/matching.py

```diff
@@ -120,35 +125,50 @@ def score_pairing(
     # (similar for prefers and skill level). It can be arbitrarily big, but must be at least bigger than the sum of
     # all the other constants used in the score.
     padded_participant_count = 10 + 2 * number_of_participants
-    base_score = 1.0
+    base_score = 5.0
     preferred_value = 3.0
     max_aux_value = 2.0
     confirmed_value = 2.0
     max_skill_value = 1.0
     required_value = padded_participant_count * sum(
         (base_score, preferred_value, max_skill_value, confirmed_value, max_aux_value)
     )
-    designated_value = required_value**2
-    unqualified_penalty = designated_value**2
+    designated_unqualified_value = required_value * required_value
+    designated_and_qualified_value = 2 * designated_unqualified_value
+    undesignated_unqualified_value = designated_unqualified_value * designated_unqualified_value
 
-    is_designated = participant in position.designated_for
     # optimally, we should reject a pairing for a participant designated for another position,
     # but we don't have that info here.
+    is_designated = participant in position.designated_for
+
+    is_qualified = position.required_skill <= participant.skill
 
-    if not is_designated and not position.required_skill <= participant.skill:
+    if not is_designated and (not is_qualified or position.designation_only):
         # the participant does not have some required skill
-        return -unqualified_penalty  # avoid matching unqualified participants
+        return -undesignated_unqualified_value  # avoid matching unqualified participants
 
     score = base_score
     if is_designated:
-        score += designated_value
+        if is_qualified:
+            score += designated_and_qualified_value
+        else:
+            # designated participants get assigned even if they don't qualify, and at a lower score
+            score += designated_unqualified_value
+
     if participant in position.preferred_by:
         score += preferred_value
     if position.required:
         score += required_value
     if participant in confirmed_participants:
         score += confirmed_value
-    score += position.skill_level * max_skill_value
+
+    if is_qualified:
+        # if qualified, lets prefer high skill positions
+        score += position.skill_level * max_skill_value
+    else:
+        # if not qualified (but designated), let's prefer low skill positions
+        score -= position.skill_level * max_skill_value
```

Comment:
this is new, and led to designated participants being put to *any* position of the designated atomic block, because being qualified wasn't taken into account in that case.